File: interview/word_count.py
import sys
from collections import Counter, defaultdict


def print_word(result: Counter):
    for word, frequency in result.most_common():
        print('<{}>: <{}>'.format(word, frequency))


def word_count():
    result = Counter()
    with open(filename, 'r') as text_file:
        for line in text_file:
            for word in line.split():
                result[word] += 1
    return result


def main():
    try:
        result = word_count()
        print_word(result)
    except IOError:
        print('File with {} does not exist'.format(filename))


def word_count_defaultdict():
    result = defaultdict(lambda: 0)
    with open(filename, 'r') as text_file:
        for line in text_file:
            for word in line.split():
                result[word] += 1
    return result


def print_word_defaultdict(result: defaultdict):
    for word, frequency in sorted(result.items(), key=lambda res: (-res[1], res[0])):
        print(f'<{word}>: <{frequency}>')


def main_defaultdict():
    try:
        result = word_count_defaultdict()
        print_word_defaultdict(result)
    except IOError:
        print('File with {} does not exist'.format(filename))


if __name__ == "__main__":
    try:
        filename = sys.argv[1]
        main()
    except IndexError:
        print('You should specify the filename')

File: interview/word_count.js
const fs = require("fs");


function word_count(filename) {
    let result = {};

    fs.readFile(filename, {encoding: 'utf-8'}, function(err, data){
    if (!err) {

        data.split('\n').forEach(
            line => line.split(' ').forEach(
                word => {
                   if (word in result) {
                       result[word] += 1
                   } else {
                       result[word] = 1
                   }
                }
            )
        );

        result = Object.entries(result).sort(
            (a, b) => {
                if (a[1] < b[1]) return 1;
                if (a[1] > b[1]) return -1;
                if (a[0].localeCompare(b[0]) === 1) return 1;
                if (b[0].localeCompare(a[0]) === -1) return -1;
            }
        );

        result.forEach(
            word => console.log(`<${word[0]}>: <${word[1]}>`)
        )
    } else {
        console.log(`File with ${filename} does not exist`);
    }
  });
}


function main() {

  let filename = process.argv.slice(2)[0];
  console.log(filename);

  word_count(filename);
}


main();
